packages/web/src/components/track/ViewerActionButtons.tsx

```diff
@@ -0,0 +1,164 @@
+import { MouseEvent, ReactNode, useCallback } from 'react'
+
+import { useGetPlaylistById, useGetTrackById } from '@audius/common/api'
+import { ID } from '@audius/common/models'
+import { gatedContentSelectors } from '@audius/common/store'
+import { Flex, Text } from '@audius/harmony'
+import { useSelector } from 'react-redux'
+
+import FavoriteButton from 'components/alt-button/FavoriteButton'
+import RepostButton from 'components/alt-button/RepostButton'
+import ShareButton from 'components/alt-button/ShareButton'
+import Tooltip from 'components/tooltip/Tooltip'
+
+import { GatedConditionsPill } from './GatedConditionsPill'
+import styles from './desktop/TrackTile.module.css'
+
+const { getGatedContentStatusMap } = gatedContentSelectors
+
+const messages = {
+  share: 'Share',
+  repost: 'Repost',
+  unrepost: 'Unrepost',
+  favorite: 'Favorite',
+  unfavorite: 'Unfavorite'
+}
+
+type ViewerActionButtonProps = {
+  contentId: ID // Collection or Track ID
+  contentType: 'track' | 'collection'
+  hasStreamAccess?: boolean
+  isDisabled?: boolean
+  isLoading?: boolean
+  rightActions?: ReactNode
+  bottomBar?: ReactNode
+  isDarkMode?: boolean
+  isMatrixMode: boolean
+  showIconButtons?: boolean
+  onClickRepost: (e?: any) => void
+  onClickFavorite: (e?: any) => void
+  onClickShare: (e?: any) => void
```

Comment:
ditto nit on better types for event handler - can this not be `MouseEvent`?

Reply:
totally can, forgot about these placeholders